dask/tests/test_task_spec.py

```diff
@@ -34,6 +33,16 @@ def clear_func_cache():
     _func_cache_reverse.clear()
 
 
+def convert_legacy_graph(*args, **kwargs):
```

Comment:
nit: I personally don't like that this shadows `convert_legacy_graph` from `dask._task_spec`. A more explicit name could highlight that this has some additional assertions baked in.